### DISCUSSION TITLE
Foreign Key as Primary Key in One-to-one relation

### DISCUSSION THREAD

**Discussion by carlocorradini** [prisma/prisma#15990] [Q&A] [answered]
I'm struggling to create a foreign key as a primary key. 
Given the example shown in the [docs](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/one-to-one-relations), I've refactored it trying to achieve the latter: 

```prisma
model User {
  id      Int     @id @default(autoincrement())
  profile Profile
}

model Profile {
  id   Int  @id
  user User @relation(fields: [id], references: [id])
}
```

However, it fails due to the following error:

*Error parsing attribute "@relation": The relation field `profile` on Model `User` is required. This is no longer valid because it's not possible to enforce this constraint on the database level. Please change the field type from `Profile` to `Profile?` to fix this.*

How can I solve this (why I can't force `profile` to be consistent without the `?`)?
Thanks 🥳

**Comment by Rackar** [comment]
one-to-one is just this:
```
model User {
  id      Int     @id @default(autoincrement())
  profile Profile
}

model Profile {
  id   Int  @id
  user User @relation(fields: [userId], references: [id])
  userId Int @unique 
}
```

**Comment by nurul3101** [comment, marked as answer]
Hello @carlocorradini 👋 

I am trying to understand your use case. The models that you have defined essentially would mean that the value of `id` in the Profile model should be the same as the `id` of the User model. Ideally, each record should have its own unique primary key which would not happen in this case as the primary key of the Profile model is dependent on the User model. In this case if both the ids are going to be the same can't you just merge both tables into one?

Also, the [relation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations) fields do not exist in the database. They are just at the Prisma level, so the fields `profile` and `user` will not exist in the database.

**Reply by carlocorradini** [reply]
@nurul3101
Thanks for your response. 
I don't want to merge them to keep the original table (e.g. `User`) cleaner and improve caching and performance (see [this](https://stackoverflow.com/a/10983016/6676781)).
Since a profile exists only for a specific user, I don't want a new primary key that differs from the one of that user (adding also a foreign key column to `User`). The primary key must be also the foreign key.
The following example works: 
```prisma
model User {
  id      Int     @id @default(autoincrement())
  profile Profile?
}

model Profile {
  id   Int  @id
  user User @relation(fields: [id], references: [id])
}
```

However, I'd like to make the `profile` in `User` **NON-NULLABLE** (without the `?`).

**Reply by nurul3101** [reply]
I think the error message suggests that it is not possible to enforce it on the database level because let's say if a particular user is deleted then the corresponding profile record would not be automatically deleted and remain dangling. But as the profile field doesn't exist on the database level it should not matter in my opinion.

**Comment by codinginflow** [comment]
I'm trying to do the same thing. I use multi-table inheritance as described in [your documentation](https://www.prisma.io/docs/orm/prisma-schema/data-model/table-inheritance#multi-table-inheritance-mti). But if the child table and parent table have different IDs, querying them becomes more difficult. This would be easier if the foreign key of the child table was the primary key and identical with the parent table's primary key.

**Reply by ludralph** [reply]
Hi @codinginflow 

Can you please open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your usecase so that our engineers can take a look?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
